TFNetworkLayer.py

```diff
@@ -1044,16 +1046,33 @@ def batch_norm(self, data,
       https://github.com/deepmind/sonnet/blob/master/sonnet/python/modules/batch_norm.py
     """
     with tf.variable_scope("batch_norm"):
+      input_data = data.copy()
```

Comment:
Why do you rename it here? And why the copy?

Reply:
The axes order might be changed in `input_data` later. But I want to keep the `data` unchanged.